internals/resource-managers/output.go

```diff
@@ -0,0 +1,133 @@
+package resourceprepares
+
+import (
+	"context"
+	"github.com/cisco-open/operator-tools/pkg/secret"
+	rcsv1alpha1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+	rclient "github.com/dana-team/container-app-operator/internals/wrappers"
+	"github.com/go-logr/logr"
+	loggingv1beta1 "github.com/kube-logging/logging-operator/pkg/sdk/logging/api/v1beta1"
+	"github.com/kube-logging/logging-operator/pkg/sdk/logging/model/output"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type OutputManager struct {
+	Ctx       context.Context
+	K8sclient client.Client
+	Log       logr.Logger
+}
+
+func (o OutputManager) prepareResource(capp rcsv1alpha1.Capp) loggingv1beta1.Output {
+	outputName := capp.GetName() + "-output"
+	if capp.Spec.LogSpec.Type == "elastic" {
```

Comment:
This function is to long and responsible on to many things. It doesn't  support the single responsibility  principle and the open close principle.
instead of this if which generates output object, ill separate to 2 different functions and make a map of string and callable the function will receive the log spec and will generate the fitting object